.github/workflows/jira.yaml

```diff
@@ -50,13 +78,61 @@ jobs:
         JIRA_USER_EMAIL: ${{ secrets.JIRA_SYNC_USER_EMAIL }}
         JIRA_API_TOKEN: ${{ secrets.JIRA_SYNC_API_TOKEN }}
     - name: Preprocess
-      if: github.event.action == 'opened' || github.event.action == 'created'
+      if: github.event.action == 'opened' || github.event.action == 'created' || github.event_name == 'workflow_dispatch'
       id: preprocess
       run: |
-        if [[ "${{ github.event_name }}" == "pull_request_target" ]]; then
+        if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+          if [ -z "${{ inputs.dispatch-gh-type }}" || -z "${{ inputs.dispatch-gh-number }}"]; then
+            echo "error: workflow_dispatch but GH no type or number provided"
+            exit 1
+          fi
+        
+          echo "type=${{ inputs.dispatch-gh-type }}" >> "$GITHUB_OUTPUT"
+          echo "gh_number=${{ inputs.dispatch-gh-number }}" >> "$GITHUB_OUTPUT"
+        
+          if [[ "${{ inputs.dispatch-gh-type }}" == "PR" ]]; then
+            TITLE=$(gh pr view ${{ inputs.dispatch-gh-number }} --json=title -q .title)
```

Comment:
TODO: I can probably make these be a single CLI call and extract values from the output independently.